<a href="https://colab.research.google.com/github/tjain858/Google-play-EDA/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

##**INTRODUCTION :**
The objective of this project is to deliver insights to understand customer demands better and thus help developers to popularize the product.It is of 10k Play Store apps for analyzing the Android market. It consists of in total of 10841 rows and 13 columns. The tools that are going to be used for this EDA would be numpy, pandas, matplotlib and seaborn which I have learnt from the course..

## Discussion of play store dataset will involve various steps such as:

*  loading the data into data frame
*  cleaning the data
*  extracting statistics from the dataset
*  exploratory analysis and visualizations
*  questions that can be asked from the dataset
*  conclusion

**Imports**: Let us start by importing some of the required libraries with which we will be working on.

In [ ]:
#imports Some importance libries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

## **Data Prepration and Cleaning:** In this section, we will be loading the Play Store Apps data stored in csv using pandas which is a fast and powerful python library for data analysis and easy data manipulation in pandas DataFrame object. We will then perform some data preparation and also cleaning on it.

* ### Load the file using pandas
* ### Look at the some information about the data & the columns
* ### Fix any missing or incorrect values

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
Dataset_url = 'https://drive.google.com/drive/folders/1O6uEhqbxf71Bk-8xz8kdKeD-Tb2YNq_d'

In [ ]:
# Loading the dataset as a pandas data frame.
playstore_df = pd.read_csv('/content/drive/MyDrive/CAPSTON PROJECT EDA/Google-play-EDA/Play Store Data.csv')

After loading the dataset, we can start the exploration but before that, we need to check and see that the dataset is ready for performing several exploration operations or not, so let’s first have a look at the structure and the manner in which the data is *organized*

In [ ]:
# look at the first 10 records in the apps dataframe
playstore_df.head(10)


App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                         Paper flowers instructions  ART_AND_DESIGN     4.4   
6            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   
7                                   Infinite Painter  ART_AND_DESIGN     4.1   
8                               Garden Coloring Book  ART_AND_DESIGN     4.4   
9                      Kids Paint Free - Drawing Fun  ART_AND_DESIGN     4.7   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   
5     167  5.6M      50,000+  Free     0       Everyone   
6     178   19M      50,000+  Free     0       Everyone   
7   36815   29M   1,000,000+  Free     0       Everyone   
8   13791   33M   1,000,000+  Free     0       Everyone   
9     121  3.1M      10,000+  Free     0       Everyone   

                      Genres        Last Updated         Current Ver  \
0               Art & Design     January 7, 2018               1.0.0   
1  Art & Design;Pretend Play    January 15, 2018               2.0.0   
2               Art & Design      August 1, 2018               1.2.4   
3               Art & Design        June 8, 2018  Varies with device   
4    Art & Design;Creativity       June 20, 2018                 1.1   
5               Art & Design      March 26, 2017                 1.0   
6               Art & Design      April 26, 2018                 1.1   
7               Art & Design       June 14, 2018            6.1.61.1   
8               Art & Design  September 20, 2017               2.9.2   
9    Art & Design;Creativity        July 3, 2018                 2.8   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up  
5    2.3 and up  
6  4.0.3 and up  
7    4.2 and up  
8    3.0 and up  
9  4.0.3 and up

In [ ]:
playstore_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [ ]:
# look at the random 10 records in the apps dataframe
playstore_df.sample(10)

App             Category  \
8011                        CW BLE Peripheral Simulator        COMMUNICATION   
8771                                         Dr. Muscle   HEALTH_AND_FITNESS   
7676                                     CP Creepypasta               FAMILY   
1113  NerdWallet: Personal Finance, Credit Score & Cash              FINANCE   
6497                      NOOK: Read eBooks & Magazines  BOOKS_AND_REFERENCE   
8698          Best DP and Status - All Type DP & Status               FAMILY   
1960                                   Metal Soldiers 2                 GAME   
6971                                  Super Jabber Jump                 GAME   
9001                                      DW Tech Tools             BUSINESS   
3110                      World Travel Guide by Triposo     TRAVEL_AND_LOCAL   

      Rating Reviews                Size     Installs  Type Price  \
8011     NaN       2                1.5M         500+  Free     0   
8771     4.0      87                 33M      10,000+  Free     0   
7676     3.8     128                2.9M      10,000+  Free     0   
1113     4.6    2417                 28M     100,000+  Free     0   
6497     4.5  155466  Varies with device  10,000,000+  Free     0   
8698     NaN       1                5.7M         100+  Free     0   
1960     4.4  153381  Varies with device  10,000,000+  Free     0   
6971     4.3   85882                 23M  10,000,000+  Free     0   
9001     NaN       1                5.9M         100+  Free     0   
3110     4.4    6012                 81M     500,000+  Free     0   

     Content Rating             Genres      Last Updated         Current Ver  \
8011       Everyone      Communication     March 8, 2017                 1.0   
8771       Everyone   Health & Fitness     July 28, 2018                 2.5   
7676           Teen      Entertainment     July 11, 2016               1.0.2   
1113       Everyone            Finance    August 3, 2018               3.5.0   
6497           Teen  Books & Reference    April 25, 2018  Varies with device   
8698       Everyone      Entertainment      May 30, 2018                 1.0   
1960           Teen             Action     June 18, 2018              1.11.2   
6971       Everyone             Arcade  November 7, 2017            6.6.3106   
9001       Everyone           Business     July 24, 2018                26.1   
3110       Everyone     Travel & Local     March 5, 2018               4.6.0   

             Android Ver  
8011          4.4 and up  
8771        4.0.3 and up  
7676        4.0.3 and up  
1113          4.4 and up  
6497  Varies with device  
8698        4.0.3 and up  
1960          4.1 and up  
6971          2.3 and up  
9001          4.1 and up  
3110        4.0.3 and up

In [ ]:
# TO know checking what columns are there in the dataset
playstore_df.columns


Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [ ]:
# look at the number of rows and columns in dataset
playstore_df.shape

(10841, 13)

In [ ]:
# look at the info of the dataframe
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


#**Description of App Dataset columns**

1.    App : The name of the app
2.    Category : The category of the app
3.    Rating : The rating of the app in the Play Store
4.    Reviews : The number of reviews of the app
5.    Size : The size of the app
6.    Install : The number of installs of the app
7.    Type : The type of the app (Free/Paid)
8.    price: The price of the app (0 if it is Free)
9.    Content Rating :The appropiate target audience of the app
10.   Genres: The genre of the app
11.   Last Updated : The date when the app was last updated
12.   Current Ver : The current version of the app
13.   Android Ver : The minimum Android version required to run the app

In [ ]:
# To check missing vlaues per columns 
playstore_df.isna().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

We saw that the dataset contains many Null or missing values. The column Rating, Type , Content Rating , Current Ver , and Android Ver contains . 1474, 1, 1, 8, and 3 missing values respectively.

Will it not be better if we can define a function to get more useful information about the different attributes of the dataset, also there is one more valid point in defining a function which it will be reusable, and we are going to utilize our defined function several times in future


In [ ]:
# Define a function to check null vlaues in dataset
def printinfo():

    '''This function return the data type , null values and unique value in dataframe
  '''
    temp = pd.DataFrame(index=playstore_df.columns)
    temp['data_type'] = playstore_df.dtypes
    temp['null_count'] = playstore_df.isnull().sum()
    temp['unique_count'] = playstore_df.nunique()
    return temp

In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

We have some useful information about the dataset. i.e., we can now see the missing number of values of any attribute, its unique count, and its respective data types.

Now we can start the process of data cleaning, lets start with the column 'Type' :-

In [ ]:
playstore_df[playstore_df.Type.isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

Since there is only one missing value in this column, So, let’s fill the missing value with "Free"

In [ ]:
playstore_df['Type'].fillna("Free",inplace= True)

After filling the value we can check and see if that has been correctly placed.

In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           0             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

Now, we can move on to the column "Content Rating" :

---



---



In [ ]:
playstore_df[playstore_df['Content Rating'].isnull()]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

By, looking only at these rows it is not easy to say what's actually missing in this row. let us have a look at all of its near rows data. For this purpose, we have iloc and loc function.

In [ ]:
playstore_df.loc[10465:10480,:]

App         Category  Rating  \
10465              Portable Wi-Fi hotspot Free    COMMUNICATION     4.0   
10466                           Wi-Fi Networks     PRODUCTIVITY     4.2   
10467                                   FI CFL          FINANCE     3.7   
10468                         Tassa.fi Finland        LIFESTYLE     3.6   
10469              TownWiFi | Wi-Fi Everywhere    COMMUNICATION     3.9   
10470                               Jazz Wi-Fi    COMMUNICATION     3.4   
10471                         Xposed Wi-Fi-Pwd  PERSONALIZATION     3.5   
10472  Life Made WI-Fi Touchscreen Photo Frame              1.9    19.0   
10473                  osmino Wi-Fi: free WiFi            TOOLS     4.2   
10474                             Sat-Fi Voice    COMMUNICATION     3.4   
10475                         Wi-Fi Visualizer            TOOLS     3.9   
10476                    Lennox iComfort Wi-Fi        LIFESTYLE     3.0   
10477              Sci-Fi Sounds and Ringtones  PERSONALIZATION     3.6   
10478                            Sci Fi Sounds           FAMILY     3.2   
10479                       Free Wi-fi HotspoT    COMMUNICATION     4.1   
10480           FJ 4x4 Cruiser Offroad Driving           FAMILY     4.1   

      Reviews    Size     Installs  Type     Price Content Rating  \
10465    1711    2.1M     100,000+  Free         0       Everyone   
10466     115    1.4M      10,000+  Free         0       Everyone   
10467     112    3.9M      10,000+  Free         0       Everyone   
10468     346    7.5M      50,000+  Free         0       Everyone   
10469    2372     58M     500,000+  Free         0       Everyone   
10470      49    4.0M      10,000+  Free         0       Everyone   
10471    1042    404k     100,000+  Free         0       Everyone   
10472    3.0M  1,000+         Free     0  Everyone            NaN   
10473  134203    4.1M  10,000,000+  Free         0       Everyone   
10474      37     14M       1,000+  Free         0       Everyone   
10475     132    2.6M      50,000+  Free         0       Everyone   
10476     552    7.6M      50,000+  Free         0       Everyone   
10477     128     11M      10,000+  Free         0       Everyone   
10478       4    8.0M       1,000+  Free         0       Everyone   
10479     382    2.3M      50,000+  Free         0       Everyone   
10480    3543     49M     500,000+  Free         0       Everyone   

                  Genres        Last Updated Current Ver   Android Ver  
10465      Communication    January 11, 2018  1.2.5.4-11    2.3 and up  
10466       Productivity     October 5, 2015         1.4    4.1 and up  
10467            Finance        July 5, 2018       1.1.1    5.0 and up  
10468          Lifestyle        May 22, 2018         5.5    4.0 and up  
10469      Communication      August 2, 2018       4.2.1    4.2 and up  
10470      Communication   February 10, 2017         0.1    2.3 and up  
10471    Personalization      August 5, 2014       3.0.0  4.0.3 and up  
10472  February 11, 2018              1.0.19  4.0 and up           NaN  
10473              Tools      August 7, 2018     6.06.14    4.4 and up  
10474      Communication   November 21, 2014     2.2.1.5    2.2 and up  
10475              Tools        May 17, 2017       0.0.9    2.3 and up  
10476          Lifestyle      March 22, 2017      2.0.15  2.3.3 and up  
10477    Personalization  September 27, 2017         4.0    4.0 and up  
10478      Entertainment    November 2, 2017         1.0    4.0 and up  
10479      Communication       July 20, 2018         2.5    4.0 and up  
10480         Simulation     January 4, 2017         1.1    2.3 and up

We can clearly see that row 10472 has missing data for the Categorycolumn and all the prevailing column values are being replaced with its previous column. A better idea will be to drop this row from our data frame.

In [ ]:
playstore_df.dropna(subset = ['Content Rating'], inplace=True)

We are having some of the unwanted columns which will be of not much use in the analysis process. So let’s drop those columns

In [ ]:
playstore_df.drop(['Current Ver','Last Updated', 'Android Ver'], axis=1, inplace=True)


Here, we realized that there are 1474 rows having null values under column 'Rating'. Hence, we decided to replace the null values with zero of overall 'Rating' values

In [ ]:
# Replace Nan values to zeros of overall 
playstore_df['Rating']=playstore_df['Rating'].fillna(0)

In [ ]:
playstore_df.tail(20)

App             Category  \
10821                                        Poop FR               FAMILY   
10822                                      PLMGSS FR         PRODUCTIVITY   
10823                                   List iptv FR        VIDEO_PLAYERS   
10824                                      Cardio-FR              MEDICAL   
10825                             Naruto & Boruto FR               SOCIAL   
10826      Frim: get new friends on local chat rooms               SOCIAL   
10827                             Fr Agnel Ambarnath               FAMILY   
10828                        Manga-FR - Anime Vostfr               COMICS   
10829                 Bulgarian French Dictionary Fr  BOOKS_AND_REFERENCE   
10830                              News Minecraft.fr   NEWS_AND_MAGAZINES   
10831                       payermonstationnement.fr  MAPS_AND_NAVIGATION   
10832                                       FR Tides              WEATHER   
10833                                    Chemin (fr)  BOOKS_AND_REFERENCE   
10834                                  FR Calculator               FAMILY   
10835                                       FR Forms             BUSINESS   
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10821     0.0       6                2.5M          50+  Free     0   
10822     0.0       0                3.1M          10+  Free     0   
10823     0.0       1                2.9M         100+  Free     0   
10824     0.0      67                 82M      10,000+  Free     0   
10825     0.0       7                7.7M         100+  Free     0   
10826     4.0   88486  Varies with device   5,000,000+  Free     0   
10827     4.2     117                 13M       5,000+  Free     0   
10828     3.4     291                 13M      10,000+  Free     0   
10829     4.6     603                7.4M      10,000+  Free     0   
10830     3.8     881                2.3M     100,000+  Free     0   
10831     0.0      38                9.8M       5,000+  Free     0   
10832     3.8    1195                582k     100,000+  Free     0   
10833     4.8      44                619k       1,000+  Free     0   
10834     4.0       7                2.6M         500+  Free     0   
10835     0.0       0                9.6M          10+  Free     0   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     0.0       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                   Genres  
10821       Everyone            Entertainment  
10822       Everyone             Productivity  
10823       Everyone  Video Players & Editors  
10824       Everyone                  Medical  
10825           Teen                   Social  
10826     Mature 17+                   Social  
10827       Everyone                Education  
10828       Everyone                   Comics  
10829       Everyone        Books & Reference  
10830       Everyone         News & Magazines  
10831       Everyone        Maps & Navigation  
10832       Everyone                  Weather  
10833       Everyone        Books & Reference  
10834       Everyone                Education  
10835       Everyone                 Business  
10836       Everyone                Education  
10837       Everyone                Education  
10838       Everyone                  Medical  
10839     Mature 17+        Books & Reference  




Finally, after fixing all the missing values, we should have a look at our data frame, We defined a function as "printinfo()". So, it’s time to use that function.


In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            40
Reviews           object           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

All the columns have the null_count as zero, which indicates that now the data frame doesn’t contain any missing values.


> ### Now we are done with the data cleansing part and in a state to start the work for data preparation




Columns like  Reviews, Size, Installs, & price  should have an int or float data type, But here we can see of object type, So let’s convert them to their respective correct type.

Starting with the column Reviews , converting its type to int .



In [ ]:
playstore_df['Reviews'] = playstore_df.Reviews.astype(int)

We can see that the changes have taken its effect or not by calling our printinfo() function.



In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            40
Reviews            int64           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

Now, the reviews column has been converted to int type, so now we can move to the Column: Size

Converting the Size Column from object to integer, but this column contains some of the special characters like , , + , M , K  and also it has a some of the value as '"aries with device" . We need to remove all of these and then convert it to int or float .

In [ ]:
playstore_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

   Reviews  Size     Installs  Type Price Content Rating  \
0      159   19M      10,000+  Free     0       Everyone   
1      967   14M     500,000+  Free     0       Everyone   
2    87510  8.7M   5,000,000+  Free     0       Everyone   
3   215644   25M  50,000,000+  Free     0           Teen   
4      967  2.8M     100,000+  Free     0       Everyone   

                      Genres  
0               Art & Design  
1  Art & Design;Pretend Play  
2               Art & Design  
3               Art & Design  
4    Art & Design;Creativity

By diagnosing the data frame, we know that:

1. Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'
2. Values of column 'Size' are strings representing size in 'M' as Megabytes,   'k' as kilobytes and also 'Varies with devices'.
3. Values of column 'Installs' are strings representing install amount with symbols such as ',' and '+'.
4. Values of column 'Price' are strings representing price with symbol '$'.


Hence, we will need to do some data cleaning.





so now we can move to the Column : **Size** & some of the special characters like  **, , + , M , K** and also it has a some of the value as **"varies with device"** . We need to remove all of these and then convert it to int or float

In [ ]:
# TO see a unique value i
playstore_df['Size'].unique()

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [ ]:
# Removing the '+' Symbol by lamda function
playstore_df['Size'] = playstore_df.Size.apply(lambda x : x.strip('+'))

# Removing the ',' symbol to '' by using lambda function
playstore_df['Size'] = playstore_df.Size.apply(lambda x :x.replace(',',''))




Since, size of the applications present in the datset are in MB and KB. Therefore, for ease in data processing, entire size column is converted to MB.

In [ ]:
# Replacing the M symbol by multplying the value with 1000000 
playstore_df['Size'] = playstore_df.Size.apply(lambda x : str(x).replace('M' , '') if 'M' in str(x) else x)


# Replacing the 'k' symbol by '' and divided by 1000 to convert it to MB
playstore_df['Size'] = playstore_df.Size.apply(lambda x :float(str(x).replace('k' , ''))/1000 if 'k' in str(x) else x ) # Formula	:- divide the digital storage value by 1000


In [ ]:
playstore_df.tail(5)

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating  Reviews                Size     Installs  Type Price  \
10836     4.5       38                  53       5,000+  Free     0   
10837     5.0        4                 3.6         100+  Free     0   
10838     0.0        3                 9.5       1,000+  Free     0   
10839     4.5      114  Varies with device       1,000+  Free     0   
10840     4.5   398307                  19  10,000,000+  Free     0   

      Content Rating             Genres  
10836       Everyone          Education  
10837       Everyone          Education  
10838       Everyone            Medical  
10839     Mature 17+  Books & Reference  
10840       Everyone          Lifestyle

In [ ]:
# Replacing the "Varies with device" value with Nan 
playstore_df['Size'] = playstore_df.Size.replace('Varies with device' , np.NaN)

In [ ]:
playstore_df.tail(10)

App             Category  \
10831                       payermonstationnement.fr  MAPS_AND_NAVIGATION   
10832                                       FR Tides              WEATHER   
10833                                    Chemin (fr)  BOOKS_AND_REFERENCE   
10834                                  FR Calculator               FAMILY   
10835                                       FR Forms             BUSINESS   
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating  Reviews   Size     Installs  Type Price Content Rating  \
10831     0.0       38    9.8       5,000+  Free     0       Everyone   
10832     3.8     1195  0.582     100,000+  Free     0       Everyone   
10833     4.8       44  0.619       1,000+  Free     0       Everyone   
10834     4.0        7    2.6         500+  Free     0       Everyone   
10835     0.0        0    9.6          10+  Free     0       Everyone   
10836     4.5       38     53       5,000+  Free     0       Everyone   
10837     5.0        4    3.6         100+  Free     0       Everyone   
10838     0.0        3    9.5       1,000+  Free     0       Everyone   
10839     4.5      114    NaN       1,000+  Free     0     Mature 17+   
10840     4.5   398307     19  10,000,000+  Free     0       Everyone   

                  Genres  
10831  Maps & Navigation  
10832            Weather  
10833  Books & Reference  
10834          Education  
10835           Business  
10836          Education  
10837          Education  
10838            Medical  
10839  Books & Reference  
10840          Lifestyle

Now, converting all this values  numeric type

In [ ]:
# Change the type 'object' to 'float'

playstore_df['Size'] = playstore_df['Size'].astype(float)

In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            40
Reviews            int64           0          6001
Size             float64        1695           459
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119